client/lib/media/test/utils.js

```diff
@@ -84,7 +85,45 @@ describe( 'MediaUtils', function() {
 		} );
 	} );
 
+	describe( '#getFileExtension()', function() {
+		useFakeDom();
+
+		it( 'should return undefined for a falsey media value', function() {
+			expect( MediaUtils.getFileExtension() ).to.be.undefined;
+		} );
+
+		it( 'should detect extension from file name', function() {
+			expect( MediaUtils.getFileExtension( 'example.gif' ) ).to.equal( 'gif' );
+		} );
+
+		it( 'should ignore querystring parameters', function() {
+			expect( MediaUtils.getFileExtension( 'example.gif?w=100' ) ).to.equal( 'gif' );
+		} );
+
+		it( 'should detect extension from HTML5 File object', function() {
+			expect( MediaUtils.getFileExtension( new window.File( [''], 'example.gif' ) ) ).to.equal( 'gif' );
+		} );
+
+		it( 'should detect extension from object file property', function() {
+			expect( MediaUtils.getFileExtension( { file: 'example.gif' } ) ).to.equal( 'gif' );
+		} );
+
+		it( 'should detect extension from already computed extension property', function() {
+			expect( MediaUtils.getFileExtension( { extension: 'gif' } ) ).to.equal( 'gif' );
+		} );
+
+		it( 'should detect extension from object URL property', function() {
+			expect( MediaUtils.getFileExtension( { URL: 'example.gif' } ) ).to.equal( 'gif' );
+		} );
+
+		it( 'should detect extension from object guid property', function() {
+			expect( MediaUtils.getFileExtension( { guid: 'example.gif' } ) ).to.equal( 'gif' );
+		} );
+	} );
+
 	describe( '#getMimePrefix()', function() {
+		useFakeDom();
```

Comment:
If the fake DOM is needed across multiple tests, we should probably move `useFakeDom` to the top-level `describe` block.